src/Jackalope/Node.php

```diff
@@ -1656,8 +1683,9 @@ public function beginTransaction()
         parent::beginTransaction();
 
         // Notify the children properties
-        foreach ($this->properties as $prop) {
-            $prop->beginTransaction();
+        foreach ($this->propertyValues as $name => $propertyValue) {
+            $property = $this->loadProperty($name);
```

Comment:
i don't think there is much missing. just revert the changes on the transaction methods and pass the transaction status in loadProperty should do the trick. the problem is that it will become hard to see what is going on once refactored this way.